app/services/authentication/providers/apple.rb

```diff
@@ -0,0 +1,63 @@
+module Authentication
+  module Providers
+    # Apple authentication provider, uses omniauth-apple as backend
+    class Apple < Provider
+      SETTINGS_URL = "https://appleid.apple.com/account/manage".freeze
+
+      def new_user_data
+        # Apple sends `first_name` and `last_name` as separate fields
+        name = "#{info.first_name} #{info.last_name}"
+        timestamp = raw_info.auth_time || raw_info.id_info.auth_time
```

Comment:
To be honest, this is something that has made me scratch my head quite a bit. I've had times where `auth_time` comes nested directly in `raw_info` and other times within `id_info`.

Because of this mismatch every now and then ran into an error with `Time.zone.at(nil)`, this will catch both cases. If anyone is absolutely certain we'll get one or the other let me know, but from my experience I rather leave it this way 😅

Same a below for `existing_user_data `

Reply:
It might be the difference between first time vs second time. I noticed the payload Apple sends the first time (so you have to disconnect yourself on Apple's dashboard) is slightly different. I probably didn't catch this one :(

Reply:
Have you gotten around to test this?